src/library_wasmfs.js

```diff
@@ -151,13 +151,42 @@ FS.createPreloadedFile = FS_createPreloadedFile;
     // TODO: stat
     // TODO: lstat
     chmod: (path, mode) => {
-      return withStackSave(() => {
+      var err = withStackSave(() => {
         var buffer = stringToUTF8OnStack(path);
         return __wasmfs_chmod(buffer, mode);
       });
+      if(-err == {{{ cDefs.ENOENT }}}) {
+        throw new Error("ENOENT");
+      } else if(-err == {{{ cDefs.EPERM }}}) {
+        throw new Error("EPERM");
```

Comment:
Perhaps the error handling could be just
```js
function handleError(err) {
  if (err < 0) {
    throw new Error(err);
  }
}
//..
handleError(err);
```
Having a handleError function could help if we decide to change what we throw later on. And for now I think throwing just the code might be enough?

Reply:
Great, I just merged that other PR to main now, so the error handling here can be simplified I think.